test/uk/gov/hmrc/plasticpackagingtax/registration/controllers/partner/RemovePartnerControllerSpec.scala

```diff
@@ -0,0 +1,165 @@
+  )
+
+  private val soleTraderPartner = aSoleTraderPartner()
+
+  private val removePartnerController = new RemovePartnerController(mockAuthAction,
+                                                                    mockJourneyAction,
+                                                                    mockRegistrationConnector,
+                                                                    mcc,
+                                                                    mockPage
+  )
+
+  override protected def beforeEach(): Unit = {
+    super.beforeEach()
+    when(mockPage.apply(any[Form[RemovePartner]], any(), any())(any(), any())).thenReturn(
+      Html("Remove Partner Page")
+    )
+  }
+
+  override protected def afterEach(): Unit = {
+    super.afterEach()
+    reset(mockRegistrationConnector, mockPage)
+  }
+
+  "Remove Partner Controller" should {
+
+    "display page when partner found in registration" in {
+      authorizedUser()
+      mockRegistrationFind(partnershipRegistration)
+
+      val result = removePartnerController.displayPage(aSoleTraderPartner().id)(getRequest())
+
+      status(result) mustBe OK
+      contentAsString(result) mustBe "Remove Partner Page"
+    }
+
+    "redirect to partner list when partner not found in registration" in {
+      authorizedUser()
+      mockRegistrationFind(partnershipRegistration)
+
+      val result = removePartnerController.displayPage(s"${soleTraderPartner.id}xxx")(getRequest())
+
+      status(result) mustBe SEE_OTHER
+      redirectLocation(result) mustBe Some(routes.PartnerListController.displayPage().url)
+    }
+
+    "remove identified partner when remove action confirmed" in {
+      authorizedUser()
+      mockRegistrationFind(partnershipRegistration)
+      mockRegistrationUpdate()
+
+      await(
+        removePartnerController.submit(soleTraderPartner.id)(
+          postJsonRequestEncoded(("value", "yes"))
+        )
+      )
+
+      val registrationCaptor: ArgumentCaptor[Registration] =
+        ArgumentCaptor.forClass(classOf[Registration])
+      verify(mockRegistrationConnector).update(registrationCaptor.capture())(any())
+
+      registrationCaptor.getValue.organisationDetails.partnershipDetails.get.partners.size mustBe 2
+      registrationCaptor.getValue.organisationDetails.partnershipDetails.get.partners.head mustBe partnershipRegistration.organisationDetails.partnershipDetails.get.partners.lift(
+        1
+      ).get
+    }
+
+    "leave registration unchanged when unknown partner id specified" in {
+      authorizedUser()
+      mockRegistrationFind(partnershipRegistration)
+      mockRegistrationUpdate()
+
+      await(
+        removePartnerController.submit(s"${soleTraderPartner.id}xxx")(
+          postJsonRequestEncoded(("value", "yes"))
+        )
+      )
+
+      verify(mockRegistrationConnector, never()).update(any())(any())
+    }
+
+    "redirect to partner list when remove confirmation is negative" in {
```

Comment:
Do we also need to check that the list of partnerships has not been modified on/after redirect here?

Reply:
It would be good to check but recreating this where the failure from the connector is mocked and the registration model captor is not invoked, probably will not check it